### ISSUE TITLE
annotate_esm error 

### ISSUE THREAD

**Issue by yotsa** [kircherlab/CADD-scripts#63]
Hi, 

I installed CADD v 1.7 and downloaded the annotations/prescored files separately and put them in the appropriate folders. I then run ./CADD.sh test/input.vcf to test the software, then I get this error 

```shell
CADD-v1.7 (c) University of Washington, Hudson-Alpha Institute for Biotechnology and Berlin Institute of Health at Charité - Univ
ersitätsmedizin Berlin 2013-2023. All rights reserved.
Running snakemake pipeline:
snakemake /tmp/tmp.5xupN9W1bE/input.tsv.gz --use-conda --conda-prefix /RUN/CADD-scripts-1.7/envs/conda --cores 1
--configfile /RUN/CADD-scripts-1.7/config/config_GRCh38_v1.7_noanno.yml --snakefile /RUN/CADD-scripts-1.7/Snakefile -q
Building DAG of jobs...
Using shell: /usr/bin/bash
Provided cores: 1 (use --cores to define parallelism)
Rules claiming more threads will be scaled down.
Job stats:
job         count
--------  -------
join            1
prepare         1
prescore        1
total           3

Select jobs to execute...
Activating conda environment: envs/conda/000a02fda4f9a3dacbe9a7c94df2b69c_
Select jobs to execute...
Activating conda environment: envs/conda/000a02fda4f9a3dacbe9a7c94df2b69c_
Removing temporary output /tmp/tmp.5xupN9W1bE/input.prepared.vcf.
Select jobs to execute...
Activating conda environment: envs/conda/fb032738807041aeaadfd1bd59628e1f_
Smartmatch is experimental at /RUN/CADD-scripts-1.7/envs/conda/fb032738807041aeaadfd1bd59628e1f_/share/ensembl-vep-110.1-0/module
s/Bio/EnsEMBL/VEP/AnnotationSource/File.pm line 472.
Removing temporary output /tmp/tmp.5xupN9W1bE/input.novel.vcf.
Select jobs to execute...
Activating conda environment: envs/conda/57cd93e005aa8f80b32c43193d51c129_
[Tue Apr 23 09:22:54 2024]
Error in rule annotate_esm:
    jobid: 10
    input: /tmp/tmp.5xupN9W1bE/input.vep.vcf.gz, data/annotations/GRCh38_v1.7/esm/esm1v_t33_650M_UR90S_1.pt, data/annotations/GRC
h38_v1.7/esm/esm1v_t33_650M_UR90S_2.pt, data/annotations/GRCh38_v1.7/esm/esm1v_t33_650M_UR90S_3.pt, data/annotations/GRCh38_v1.7/
esm/esm1v_t33_650M_UR90S_4.pt, data/annotations/GRCh38_v1.7/esm/esm1v_t33_650M_UR90S_5.pt, data/annotations/GRCh38_v1.7/esm/pep.1
10.fa
    output: /tmp/tmp.5xupN9W1bE/input.esm_missens.vcf.gz, /tmp/tmp.5xupN9W1bE/input.esm_frameshift.vcf.gz, /tmp/tmp.5xupN9W1bE/in
put.esm.vcf.gz
    log: /tmp/tmp.5xupN9W1bE/input.annotate_esm.log (check log file(s) for error details)
    conda-env: /RUN/CADD-scripts-1.7/envs/conda/57cd93e005aa8f80b32c43193d51c129_
    shell:

        model_directory=`dirname data/annotations/GRCh38_v1.7/esm/esm1v_t33_650M_UR90S_1.pt`;
        model_directory=`dirname $model_directory`;

        python /RUN/CADD-scripts-1.7/src/scripts/lib/tools/esmScore/esmScore_missense_av_fast.py         --input /tmp/tmp.5xupN9W
1bE/input.vep.vcf.gz         --transcripts data/annotations/GRCh38_v1.7/esm/pep.110.fa         --model-directory $model_directory
 --model esm1v_t33_650M_UR90S_1  --model esm1v_t33_650M_UR90S_2  --model esm1v_t33_650M_UR90S_3  --model esm1v_t33_650M_UR90S_4
--model esm1v_t33_650M_UR90S_5          --output /tmp/tmp.5xupN9W1bE/input.esm_missens.vcf.gz --batch-size 1 &> /tmp/tmp.5xupN9W1
bE/input.annotate_esm.log

        python /RUN/CADD-scripts-1.7/src/scripts/lib/tools/esmScore/esmScore_frameshift_av.py         --input /tmp/tmp.5xupN9W1bE
/input.esm_missens.vcf.gz         --transcripts data/annotations/GRCh38_v1.7/esm/pep.110.fa         --model-directory $model_dire
ctory --model esm1v_t33_650M_UR90S_1  --model esm1v_t33_650M_UR90S_2  --model esm1v_t33_650M_UR90S_3  --model esm1v_t33_650M_UR90
S_4  --model esm1v_t33_650M_UR90S_5          --output /tmp/tmp.5xupN9W1bE/input.esm_frameshift.vcf.gz --batch-size 1 &>> /tmp/tmp
.5xupN9W1bE/input.annotate_esm.log

        python /RUN/CADD-scripts-1.7/src/scripts/lib/tools/esmScore/esmScore_inFrame_av.py         --input /tmp/tmp.5xupN9W1bE/in
put.esm_frameshift.vcf.gz         --transcripts data/annotations/GRCh38_v1.7/esm/pep.110.fa         --model-directory $model_dire
ctory --model esm1v_t33_650M_UR90S_1  --model esm1v_t33_650M_UR90S_2  --model esm1v_t33_650M_UR90S_3  --model esm1v_t33_650M_UR90
S_4  --model esm1v_t33_650M_UR90S_5          --output /tmp/tmp.5xupN9W1bE/input.esm.vcf.gz --batch-size 1 &>> /tmp/tmp.5xupN9W1bE
/input.annotate_esm.log

        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
Complete log: .snakemake/log/2024-04-23T092250.542307.snakemake.log
``` 
Can you diagnose what's wrong and how to solve this? 

Thank you
Yot

**Comment by visze** [comment]
@yotsa To figure out what went wrong we need the log of the rule that failed. In your case it is rule `annotate_esm` `/tmp/tmp.5xupN9W1bE/input.annotate_esm.log`. Log might be already deleted because in the tmp directory. Maybe rerun again.

@zihhuafang Not sure if it is the same error because right now we do not know the exact caus eof the faile. Just that the rule exited unexpected. About you error: seems that the conda environment is not correct. Did you use conda or mamba? (recommend to use mamba). And the channel priority have to be ste to strict.

**Comment by yotsa** [comment]
Max, Thanks for the reply. How do I prevent the log file getting deleted after a failed run though?


Yot

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by visze** [comment]
Don't use the CADD.sh script :-) 

You can run snakemake seperately as described also in the readme. E.g. via `snakemake test/input.tsv.gz --use-conda --conda-prefix envs/conda --configfile config/config_GRCh38_v1.7.yml --snakefile Snakefile -c 1`

**Comment by raghvendra44** [comment]
So what's the final fix? did any of you figure out?
let me know guys, Even i am facing the same issue!

**Comment by raghvendra44** [comment]
> I confirmed that I use mamba and set the channel priority to strict.

how did you set the priority? and confirm that you are using mamba itself?

**Comment by raghvendra44** [comment]
Even my log file print the same error.
```
Traceback (most recent call last):
  File "/gpfs/data/user/raghvendra/CADD-1.7/CADD-scripts/src/scripts/lib/tools/esmScore/esmScore_missense_av_fast.py", line 17, in <module>
    import torch
  File "/gpfs/data/user/raghvendra/CADD-1.7/CADD-scripts/envs/conda/e1f044d61505834b60c1d2fffa15eda7_/lib/python3.10/site-packages/torch/__init__.py", line 218, in <module>
    from torch._C import *  # noqa: F403
ImportError: /gpfs/data/user/raghvendra/CADD-1.7/CADD-scripts/envs/conda/e1f044d61505834b60c1d2fffa15eda7_/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so: undefined symbol: iJIT_NotifyEvent
```

not sure how to go ahead!
any suggestions?

**Comment by ThorbenMaa** [comment]
seems to be a recent bug with torch and mkl 2024.1.0 (new mkl release from 26th of March), see here: https://github.com/pytorch/pytorch/issues/123097 and here https://pypi.org/project/mkl/#history.

Apparently downgrading to mkl 2024.0.0 solves it for now. Can you try this by manually going into the respective environment (`envs/conda/57cd93e005aa8f80b32c43193d51c129_` in your case)? I guess they will fix it with the next mkl release :-)

**Comment by raghvendra44** [comment]
> seems to be a recent bug with torch and mkl 2024.1.0 (new mkl release from 26th of March), see here: [pytorch/pytorch#123097](https://github.com/pytorch/pytorch/issues/123097) and here https://pypi.org/project/mkl/#history.
> 
> Apparently downgrading to mkl 2024.0.0 solves it for now. Can you try this by manually going into the respective environment (`envs/conda/57cd93e005aa8f80b32c43193d51c129_` in your case)? I guess they will fix it with the next mkl release :-)

Hey, I tried convert the mkl 2024.1.0 to 2024.0.0
Initially I faced a lot of issues, then after deleting the mkl 2024.1.0 lead to deleting its dependencies as well. And then upon installing the 2024.0.0 manually helped but then since the dependencies were deleted, I had to restore them (I was able to do that coz I had noted the dependencies and its versions before deleting them). Doing all this was quite an exercise but finally it worked.

After all this, I faced another issue! Which has been raised and closed i.e mmsplice1.0.1 but hasn't been resolved yet. The issue is that installing mmsplice1.0.1 has too many issues with its dependency libraries and upon installing the latest version of mmsplice does work out at the end but the predicted values arent right.

**Comment by visze** [comment]
I released a new CADD-scripts version [v1.7.1](https://github.com/kircherlab/CADD-scripts/releases/tag/v1.7.1). Maybe you try that one.  Now it is recommended to use apptainer/singularity and all environments are packed within a container and no conda builds are needed (container is 17GB large).  You also need now snakemake 8.

Also I updated the environments. So If you use mamba/conda instead I hope you will not face the issues you had above
